### DISCUSSION TITLE
Run stale on remote/other repo

### DISCUSSION THREAD

**Discussion by jcook-uptycs** [actions/stale#1012] [General]
I'm looking for a way to run this stale tool and give it a list of repos to run on. This would require a token with access to all the repo. But would not require creating a workflow on every repo. 

I have a use case where I need to run stale for many repos at the same time. This time is not set by a schedule, but a manual action. It would be nice to be able to run this from the manual action.

**Comment by gr2m** [comment]
I was just about to post a feature request for this. The way this would work is by supporting GitHub App authentication as an alternative to `repo-token`.

### In detail

You would register a GitHub App and install it on the repositories that you want the stale action to check. It does not matter where the actual stale action is run, because instead of passing the `repo-token` you would pass `app-id` and `app-private-key`, the credentials you received when you registered the app.

The stale app would then iterate through all repositories it is installed on and run the check on each repository the same way it does now.

I've built the @octokit SDKs that would make this all work and that `@actions/github` is using internally. I'm happy to send a pull request for this change. I've always meant to since it's the last piece that was missing from the stale app that the action does not support. This change would close the gap.

**Reply by benoittgt** [reply]
Maybe this could interest you? https://github.com/actions/stale/discussions/1089#discussioncomment-7239931

**Reply by gr2m** [reply]
you can also run it across accounts using a matrix and the create-github-app-token action, see https://github.com/actions/create-github-app-token#create-tokens-for-multiple-user-or-organization-accounts
